### PR TITLE
fix race condition by only going to next step in onboarding until pro…

### PR DESCRIPTION
**Pull request by NilsBacke**
…mise is complete

### CODE REVIEW

File: frontend/src/Onboarding/TransferCoursesScreen.tsx [hunk 1]
```diff
@@ -50,41 +35,6 @@ const TitleText = styled.div`
...
-  }
-  &::-webkit-scrollbar {
-    -webkit-appearance: none;
-    width: 16px;
-  }
-  &::-webkit-scrollbar-track:vertical {
-    border-left: 1px solid #e7e7e7;
-    box-shadow: 1px 0 1px 0 #f6f6f6 inset, -1px 0 1px 0 #f6f6f6 inset;
-  }
-`;
```
Comment (NilsBacke) [inline frontend/src/Onboarding/TransferCoursesScreen.tsx:86]: Remove unused code

File: frontend/src/advising/ManageStudents.tsx [hunk 2]
```diff
@@ -115,10 +113,9 @@ const StudentsList = (props: StudentsListProps) => {
   const [isLoading, setIsLoading] = useState(true);
   const [pageNumber, setPageNumber] = useState(0);
   const [isLastPage, setIsLastPage] = useState(false);
-  let token = useSelector((state: AppState) => getTokenFromState(state));
```
Comment (NilsBacke) [inline frontend/src/advising/ManageStudents.tsx:118]: Import error from previous remove token from redux state PR

**Comment by NilsBacke** [comment]
[Fix race condition in onboarding flow](https://trello.com/c/6zImQdUb/214-fix-race-condition-in-onboarding-flow)

File: frontend/src/Onboarding/GenericOnboarding.tsx [hunk 3]
```diff
@@ -198,13 +198,24 @@ const OnboardingSelectionTemplate: React.FC<
       >
         {children}
       </Paper>
-      <Link to={to} onClick={onSubmit} style={{ textDecoration: "none" }}>
+      <div
+        onClick={() => {
+          onSubmit().then(() => {
```
Comment (ArunJ98) [inline frontend/src/Onboarding/GenericOnboarding.tsx:203]: why does onSubmit need to be a promise here?

**Review by ArunJ98 — APPROVED** [review]

**Pull request merged**